historysync.go

```diff
@@ -263,42 +380,21 @@ func (portal *Portal) sendBackfill(backfillID string, messages []*imessage.Messa
 			}
 		}
 	}
-	for i, meta := range metas {
-		idMap[meta.GUID] = append(idMap[meta.GUID], eventIDs[i])
-	}
-	txn, err := portal.bridge.DB.Begin()
-	if err != nil {
-		portal.log.Errorln("Failed to start transaction to save batch messages:", err)
-		return true
-	}
-	portal.log.Debugfln("Inserting %d event IDs to database to finish backfill %s", len(eventIDs), backfillID)
-	portal.finishBackfill(txn, eventIDs, metas)
-	portal.Update(txn)
-	err = txn.Commit()
-	if err != nil {
-		portal.log.Errorln("Failed to commit transaction to save batch messages:", err)
-	}
-	portal.log.Infofln("Finished backfill %s", backfillID)
-	return true
+	return eventIDs, nil
 }
 
 func (portal *Portal) finishBackfill(txn dbutil.Transaction, eventIDs []id.EventID, metas []messageWithIndex) {
 	for i, info := range metas {
 		if info.Tapback != nil {
-			if info.Tapback.Remove {
```

Comment:
This was the reason I had asked for testing with other bridges. I had assumed that since the tapback was removed that no removal tapback or tapback in general would be in the iMessage database either by BlueBubbles or other connectors because the reaction would no longer have existed. Upon testing, the BlueBubbles connector has removal tapbacks borked but also the Tapback object in backfills doesn't have the Remove bool set to true (probably also just a BB connector problem). I'm gonna poke around for this now.

Reply:
Checking that the Type.Emoji() method doesn't return empty solves this but should also be handled correctly elsewhere when sending.

Reply:
BB is borked as in it should be sending updates that tapbacks are now null and isn't? Or the payload doesn't fit the pattern of being null? If you're not sure then I might be able to test that for you.

Reply:
So question, if I close my bridge and someone, or I remove a tapback elsewhere...the backfill is what will handle that correct? If so then I think we should try and resolve that with this PR. If not then we can resolve this comment and move on.

Reply:
> BB is borked as in it should be sending updates that tapbacks are now null and isn't? Or the payload doesn't fit the pattern of being null? If you're not sure then I might be able to test that for you.

When converting a BB message to the iMessage struct, AssociatedMessageType is passed into TapbackFromName() to create the tapback object. The issue is that TapbackFromName case switches strictly with the name but BB passes the name with a dash in front if it is removing the reaction. The function would return 0 for the type and an empty string for the emoji. This would also never set the remove flag to true.

> So question, if I close my bridge and someone, or I remove a tapback elsewhere...the backfill is what will handle that correct? If so then I think we should try and resolve that with this PR. If not then we can resolve this comment and move on.

This is handled and works as expected now! Again, I'm unsure how other connectors handle this but I will test them out today. Currently, the bridge will ignore tapbacks with removals in order to not add them (as if removing them, I know this sounds stupid but seems fine).